File: src/main/java/io/github/mdsimmo/bomberman/events/BmPlayerHurtIntent.java
package io.github.mdsimmo.bomberman.events;

import io.github.mdsimmo.bomberman.game.Game;
import io.github.mdsimmo.bomberman.utils.RefectAccess;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;

import javax.annotation.Nonnull;

/**
 * Called whenever a bm player takes damage.
 */
public class BmPlayerHurtIntent extends BmIntentCancellable {

    public static void run(Game game, Player player, Player cause) {
        var hurtEvent = new BmPlayerHurtIntent(game, player, cause);
        Bukkit.getPluginManager().callEvent(hurtEvent);
        hurtEvent.verifyHandled();
    }

    private static final HandlerList HANDLERS_LIST = new HandlerList();
    private final Game game;
    private final Player player, attacker;

    public BmPlayerHurtIntent(Game game, Player player, Player attacker) {
        this.game = game;
        this.player = player;
        this.attacker = attacker;
    }

    @Override
    @RefectAccess
    @Nonnull
    public HandlerList getHandlers() {
        return HANDLERS_LIST;
    }

    public static HandlerList getHandlerList() {
        return HANDLERS_LIST;
    }

    public Game getGame() {
        return game;
    }

    public Player getPlayer() {
        return player;
    }

    public Player getAttacker() {
        return attacker;
    }

}

File: src/main/java/io/github/mdsimmo/bomberman/events/BmPlayerHurtIntent.kt
package io.github.mdsimmo.bomberman.events

import io.github.mdsimmo.bomberman.game.Game
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.event.HandlerList

/**
 * Called whenever a bm player takes damage.
 */
class BmPlayerHurtIntent(val game: Game, val player: Player, val attacker: Player) : BmEvent(),
        IntentCancellable by BmIntentCancellable() {

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        @JvmStatic
        fun run(game: Game, player: Player, cause: Player) {
            val hurtEvent = BmPlayerHurtIntent(game, player, cause)
            Bukkit.getPluginManager().callEvent(hurtEvent)
            hurtEvent.verifyHandled()
        }

        @JvmStatic
        val handlerList = HandlerList()
    }

}